### ISSUE TITLE
Harmonizer: normalizePitch returns only repetition of one value, given a diverse data set

### ISSUE THREAD

**Issue by MalteRichert** [Rex2002/sound-investments#57]
Got the following output upon calling Harmonizer.harmonize() with TestDoubles.json as pitch data. The data don't look like they should lead to only one note repeated.

`InstrumentData(volume=[1.0], pitch=[83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83, 83], instrument=Synth 1, delayEcho=null, feedbackEcho=null, delayReverb=null, feedbackReverb=null, filterData=null, pan=[0.0])
`

**Comment by ArtInLines** [comment]
Just as a note on how the test data was created. I took the prices of (I think SAP) over some timespan and clamped them from 0 to 1 (inclusive). Meaning that the lowest price in became 0, the highest price became 1 and everything else was in-between.

I think that's how we wanted to calculate the absolute prices as doubles, but maybe not

**Comment by MalteRichert** [comment]
That sounds exactly right. I'm quite certain that it's an error in the Harmonizer algorithm, after all that code hasn't been tested before and apparently it obliges by Stroetmann's Rule ("No code more complex than Hello World can ever be without error when it hasn't been tested.").
I'll take a look at it later today.

**Comment by Rex2002** [comment]
found the problem: 
this:

```java
for (int offsetIndex = 0; offsetIndex < scale.length; offsetIndex++) {
                sum += scale[offsetIndex];
                if (pitch[i] <= sum) {
                    output[i] = 36 + offsetIndex;
                }
            }
``` 

should be this:
```java            for (int offsetIndex = 0; offsetIndex < scale.length; offsetIndex++) {
                sum += scale[offsetIndex];
                if (pitch[i] <= sum) {
                    output[i] = 36 + offsetIndex;
                    break;
                }
            }
```
because after the first time `pitch[i] <= sum` is true, it will always remain true and when we do not leave the `for(int offsetIndex...`-loop, it just all gets set to the highest value (hope that makes sense)

**Comment by nichtLehdev** [comment]
Yeah, nasty bug.. Always these problems with loops and conditions xD 

I think you could just create a quick bugfix branch as this is normally the way to go (at least I learned it that way)

**Comment by MalteRichert** [comment]
Nice catch! @Rex2002 You could also just add this to #55 by pushing to that branch (time-beat-mapping), then we can just squash it all into one commit on main

**Comment by Rex2002** [comment]
fixed with 6ed86655ef85e568b07deec1e9aeef9b58ae0396 to #55
